StackExchange.Redis/StackExchange/Redis/RedisDatabase.cs

```diff
@@ -1579,6 +1579,605 @@ public IEnumerable<SortedSetEntry> SortedSetScan(RedisKey key, RedisValue patter
+                maxId);
+
+            return ExecuteSync(msg, ResultProcessor.SingleStream);
+        }
+
+        public RedisStreamEntry[] StreamRange(RedisKey key, RedisValue minId, RedisValue maxId, int count, CommandFlags flags = CommandFlags.None)
+        {
+            if (count <= 0)
+            {
+                throw new ArgumentException("'count' must be greater than 0.");
+            }
+
+            var msg = Message.Create(Database,
+                flags,
+                RedisCommand.XRANGE,
+                key,
+                new RedisValue[]
+                {
+                    minId,
+                    maxId,
+                    StreamConstants.Count,
+                    count
+                });
+
+            return ExecuteSync(msg, ResultProcessor.SingleStream);
+        }
+
+        public Task<RedisStreamEntry[]> StreamRangeAsync(RedisKey key, RedisValue minId, RedisValue maxId, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database,
+                flags,
+                RedisCommand.XRANGE,
+                key,
+                new RedisValue[]
+                {
+                    minId,
+                    maxId
+                });
+
+            return ExecuteAsync(msg, ResultProcessor.SingleStream);
+        }
+
+        public Task<RedisStreamEntry[]> StreamRangeAsync(RedisKey key, RedisValue minId, RedisValue maxId, int count, CommandFlags flags = CommandFlags.None)
+        {
+            if (count <= 0)
+            {
+                throw new ArgumentException("'count' must be greater than 0.");
+            }
+
+            var msg = Message.Create(Database,
+                flags,
+                RedisCommand.XRANGE,
+                key,
+                new RedisValue[]
+                {
+                    minId,
+                    maxId,
+                    StreamConstants.Count,
+                    count
+                });
+
+            return ExecuteAsync(msg, ResultProcessor.SingleStream);
+        }
+
+        public RedisStreamEntry[] StreamRangeReverse(RedisKey key, RedisValue maxId, RedisValue minId, int count, CommandFlags flags = CommandFlags.None)
+        {
+            // Example: > XRANGE somestream + - COUNT 1
+
+            if (count <= 0)
+            {
+                throw new ArgumentException("'count' must be greater than 0.");
+            }
+
+            var msg = Message.Create(Database,
+                flags,
+                RedisCommand.XREVRANGE,
+                key,
+                new RedisValue[]
+                {
+                    maxId,
+                    minId,
+                    StreamConstants.Count,
+                    count
+                });
+
+            return ExecuteSync(msg, ResultProcessor.SingleStream);
+        }
+
+        public RedisStreamEntry[] StreamRangeReverse(RedisKey key, RedisValue maxId, RedisValue minId, CommandFlags flags = CommandFlags.None)
```

Comment:
Same as above - by naming `count` nullable, we can collapse overloads here and simplify

Overall: for API consistency instead of `StreamRangeReverse`, look at how `SortedSetRangeByRank` and others work - add a `Order order = Order.Ascending` argument in the same way - this again halves our overloads needed :)